From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Doe <john.doe@somewhere.on.planet>
Date: Sun, 28 Jan 2024 20:29:14 +0000
Subject: Patching kernel rockchip64 files drivers/tty/serial/8250/8250_dma.c

Signed-off-by: John Doe <john.doe@somewhere.on.planet>
---
 drivers/tty/serial/8250/8250_dma.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/tty/serial/8250/8250_dma.c b/drivers/tty/serial/8250/8250_dma.c
index 7fa66501792d..247468ce448e 100644
--- a/drivers/tty/serial/8250/8250_dma.c
+++ b/drivers/tty/serial/8250/8250_dma.c
@@ -49,19 +49,20 @@ static void __dma_rx_complete(struct uart_8250_port *p)
 	/*
 	 * New DMA Rx can be started during the completion handler before it
 	 * could acquire port's lock and it might still be ongoing. Don't to
 	 * anything in such case.
 	 */
+        dma->rx_running = 0;
 	dma_status = dmaengine_tx_status(dma->rxchan, dma->rx_cookie, &state);
-	if (dma_status == DMA_IN_PROGRESS)
-		return;
+	//if (dma_status == DMA_IN_PROGRESS)
+	//	return;
 
 	count = dma->rx_size - state.residue;
 
 	tty_insert_flip_string(tty_port, dma->rx_buf, count);
 	p->port.icount.rx += count;
-	dma->rx_running = 0;
+	//dma->rx_running = 0;
 
 	tty_flip_buffer_push(tty_port);
 }
 
 static void dma_rx_complete(void *param)
-- 
Created with Armbian build tools https://github.com/armbian/build
